### PR TITLE
UI: Provide new api service for orion-design

### PR DESCRIPTION
**Pull request by pleek91**
# Description
Part of a multi step effort to replace our current ui api services with a much simpler and more flexible approach. 

Related to https://github.com/PrefectHQ/orion-design/pull/545

### CODE REVIEW

File: orion-ui/src/App.vue [hunk 1]
```diff
@@ -12,13 +12,7 @@
...
     <ContextSidebar v-if="showMenu" class="app__sidebar" @click="close" />
     <suspense>
-      <router-view class="app__router-view">
-        <template #default="{ Component }">
-          <transition name="app__router-view-fade" mode="out-in">
-            <component :is="Component" />
-          </transition>
-        </template>
-      </router-view>
+      <RouterView />
```
Comment (stackoverfloweth) [inline orion-ui/src/App.vue:15]: I like this change but I'd use a different name. It's too easy to assume that's the same router view component provided by vue-router

Comment (pleek91) [reply orion-ui/src/App.vue:15]: Open to alternatives, any suggestions?

Comment (stackoverfloweth) [reply orion-ui/src/App.vue:15]: AppRouterView, RootRouterView, ProvideRouterView?

Comment (pleek91) [reply orion-ui/src/App.vue]: Went with `AppRouterView`

**Review by stackoverfloweth — APPROVED** [review]

**Pull request merged**